x/ccv/consumer/keeper/relay.go

```diff
@@ -194,17 +189,28 @@ func (k Keeper) SendPendingSlashRequests(ctx sdk.Context) {
 	k.ClearPendingSlashRequests(ctx)
 }
 
+// OnAcknowledgementPacket handles acknowledgments for sent VSCMatured and Slash packets
 func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, ack channeltypes.Acknowledgement) error {
 	if err := ack.GetError(); err != "" {
-		// packet was sent to a non-established channel
-		if err != sdkerrors.Wrap(
-			channeltypes.ErrInvalidChannelState,
-			packet.DestinationChannel,
-		).Error() {
-			return fmt.Errorf(err)
+		// Reasons for ErrorAcknowledgment
+		//  - packet data could not be successfully decoded
+		//	- packet sent on a non-established channel
+		//  - the Slash packet was ill-formed (errors while handling it)
+		// None of these should ever happen.
+		err := k.ChanCloseInit(ctx, packet.SourcePort, packet.SourceChannel)
+		if err != nil {
+			return fmt.Errorf("ChanCloseInit(%s) failed: %s", packet.SourceChannel, err.Error())
+		}
+		// check if there is an established CCV channel
+		channelID, found := k.GetProviderChannel(ctx)
+		if !found {
+			return sdkerrors.Wrapf(types.ErrNoProposerChannelId, "recv ErrorAcknowledgement on non-established channel %s", packet.SourceChannel)
+		}
+		if channelID != packet.SourceChannel {
+			// Close the established CCV channel as well
+			return k.ChanCloseInit(ctx, types.PortID, channelID)
```

Comment:
Why are we closing the established channel here as well?

Reply:
If we get here, the consumer chain should be shut down. The only way to do that is to close the CCV channel and then wait for the next BeginBlock where the code will panic (see [here](https://github.com/cosmos/interchain-security/blob/61153cd6084b0ddb6deae06c3a1882a5d3a481ad/x/ccv/consumer/module.go#L157)).